src/olympia/addons/serializers.py

```diff
@@ -591,3 +596,46 @@ class Meta:
         fields = ('id', 'current_version', 'default_locale',
                   'locale_disambiguation', 'name', 'target_locale', 'type',
                   'url', )
+
+
+class ReplacementAddonSerializer(serializers.ModelSerializer):
+    replacement = serializers.SerializerMethodField()
+    ADDON_PATH_REGEX = r"""/addon/(?P<addon_id>[^/<>"']+)/$"""
+    COLLECTION_PATH_REGEX = (
+        r"""/collections/(?P<user_id>[^/<>"']+)/(?P<coll_slug>[^/]+)/$""")
+
+    class Meta:
+        model = ReplacementAddon
+        fields = ('guid', 'replacement')
+
+    def _get_addon_guid(self, addon_id):
+        try:
+            addon = Addon.objects.public().id_or_slug(addon_id).get()
+        except Addon.DoesNotExist:
+            return []
+        return [addon.guid]
+
+    def _get_collection_guids(self, user_id, collection_slug):
+        try:
+            get_args = {'slug': collection_slug, 'listed': True}
+            if isinstance(user_id, basestring) and not user_id.isdigit():
+                get_args.update(**{'author__username': user_id})
+            else:
+                get_args.update(**{'author': user_id})
+            collection = Collection.objects.get(**get_args)
+        except Collection.DoesNotExist:
+            return []
+        valid_q = Addon.objects.get_queryset().valid_q()
```

Comment:
I was confused initially and thought this allowed non-public add-ons to be returned... IMHO `valid_q()` should not have a default add-on status and all callers should provide one...

Reply:
yeah... everywhere else it's used (now, anyway) supplies statuses.  I'll add one.

Reply:
Also, once old collections views are deleted (post frontend) the function can be simplified a lot - the `prefix` parameter is only used in `collection_detail`